rnd/autogpt_builder/src/components/CustomNode.tsx

```diff
@@ -4,23 +4,25 @@ import 'reactflow/dist/style.css';
 import './customnode.css';
 import InputModalComponent from './InputModalComponent';
 import OutputModalComponent from './OutputModalComponent';
-import { BlockSchema } from '@/lib/types';
+import { BlockIORootSchema, NodeExecutionResult } from '@/lib/autogpt-server-api/types';
 import { beautifyString } from '@/lib/utils';
 import { Switch } from "@/components/ui/switch"
 import NodeHandle from './NodeHandle';
 import NodeInputField from './NodeInputField';
 
-type CustomNodeData = {
+export type CustomNodeData = {
   blockType: string;
   title: string;
-  inputSchema: BlockSchema;
-  outputSchema: BlockSchema;
+  inputSchema: BlockIORootSchema;
+  outputSchema: BlockIORootSchema;
   hardcodedValues: { [key: string]: any };
   setHardcodedValues: (values: { [key: string]: any }) => void;
```

Comment:
I somehow don't like this because you need to pass function for each but I did the same with `setErrors` for validation and don't have idea how to do it well.